gatekeeper/src/main/java/com/application/gatekeeper/service/ResidentServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+import com.application.gatekeeper.request.VisitorRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.sql.Date;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Optional;
+
+import static com.application.gatekeeper.constants.CommonConstants.ROLE_GATEKEEPER;
+import static com.application.gatekeeper.constants.CommonConstants.ROLE_RESIDENT;
+import static com.application.gatekeeper.constants.ErrorConstants.ALREADY_BLACKLISTED;
+import static com.application.gatekeeper.constants.ErrorConstants.GATEKEEPER_CANNOT_CREATE_VISITOR;
+import static com.application.gatekeeper.constants.ErrorConstants.NO_SUCH_VISITOR_EXISTS;
+import static com.application.gatekeeper.constants.ErrorConstants.TIME_OVER;
+import static com.application.gatekeeper.constants.ErrorConstants.USER_DEACTIVATED;
+import static com.application.gatekeeper.constants.ErrorConstants.USER_NOT_APPROVED;
+import static com.application.gatekeeper.constants.ErrorConstants.VISITOR_ALREADY_EXISTS;
+import static com.application.gatekeeper.constants.ErrorConstants.VISITOR_BLOCKED;
+import static com.application.gatekeeper.constants.ErrorConstants.VISITOR_BUSY;
+
+@Service
+public class ResidentServiceImpl implements ResidentService{
+
+    private VisitorRepository visitorRepository;
+    private VisitorDetailsRepository visitorDetailsRepository;
+    private BlacklistRepository blacklistRepository;
+    private UserRepository userRepository;
+    private final JwtService jwtService;
+
+    @Autowired
+    public ResidentServiceImpl(VisitorRepository visitorRepository, VisitorDetailsRepository visitorDetailsRepository, BlacklistRepository blacklistRepository, UserRepository userRepository, JwtService jwtService) {
+        this.visitorRepository = visitorRepository;
+        this.visitorDetailsRepository = visitorDetailsRepository;
+        this.blacklistRepository = blacklistRepository;
+        this.userRepository = userRepository;
+        this.jwtService = jwtService;
+    }
+
+    Optional<Visitor> checkVisitorExists(String email){
+        Optional<Visitor> visitor = visitorRepository.findByEmail(email);
+        return visitor;
+    }
+
+    void checkUserApproval(){
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        Optional<User> user = userRepository.findUserByEmail(userDetails.getUsername());
+        if (! user.get().isApproved())
+            throw new GateKeeperApplicationException(USER_NOT_APPROVED);
+        if (! user.get().isActive())
```

Comment:
if user is not active it should be handle at security layer
same goes for approval as well
if the user is deleted or not yet approve the requests should not be allowed to hit controller